### ISSUE TITLE
[BUG/Help] is the new `get_position_ids` do not need gMask in 2D?

### ISSUE THREAD

**Issue by hscspring** [THUDM/ChatGLM-6B#498]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

```python
def get_position_ids(self, input_ids, mask_positions, device, gmask=False):
        batch_size, seq_length = input_ids.shape
        context_lengths = [seq.tolist().index(self.config.bos_token_id) for seq in input_ids]
        if self.position_encoding_2d:
            position_ids = torch.arange(seq_length, dtype=torch.long, device=device).unsqueeze(0).repeat(batch_size, 1)
            for i, context_length in enumerate(context_lengths):
                position_ids[i, context_length:] = mask_positions[i]
            block_position_ids = [torch.cat((
                torch.zeros(context_length, dtype=torch.long, device=device),
                torch.arange(seq_length - context_length, dtype=torch.long, device=device) + 1
            )) for context_length in context_lengths]
            block_position_ids = torch.stack(block_position_ids, dim=0)
            position_ids = torch.stack((position_ids, block_position_ids), dim=1)
        else:
            position_ids = torch.arange(seq_length, dtype=torch.long, device=device).unsqueeze(0).repeat(batch_size, 1)
            if not gmask:
                for i, context_length in enumerate(context_lengths):
                    position_ids[context_length:] = mask_positions[i]

        return position_ids

```

gMASK is not need in 2d config, that's not consistent with the last version.

### Steps To Reproduce

not refered

### Environment

```markdown
- OS:
- Python:
- Transformers:
- PyTorch:
- CUDA Support (`python -c "import torch; print(torch.cuda.is_available())"`) :
```

**Comment by duzx16** [comment]
Yes. That is expected behaviour.

**Comment by hscspring** [comment]
tks. but if so, why `else` need `gmask`.

is that means, if `position_encoding_2d` is used, it doesn't care whether using `gmask` or not?

in this `use_gmask = True if gMASK in input_ids else False` line, 
`use_gmask` should be `True`, we mask the position with a mask token for 2d encoding but not for 1d encoding.

that's really confusing...

**Comment by duzx16** [comment]
> tks. but if so, why `else` need `gmask`.
> 
> is that means, if `position_encoding_2d` is used, it doesn't care whether using `gmask` or not?
> 
> in this `use_gmask = True if gMASK in input_ids else False` line, `use_gmask` should be `True`, we mask the position with a mask token for 2d encoding but not for 1d encoding.
> 
> that's really confusing...

Yes. if `position_encoding_2d ` is True, `use_gmask` doesn't influence the  position id.
